coil-video/src/main/java/coil/decode/VideoFrameDecoderDelegate.kt

```diff
@@ -0,0 +1,157 @@
+package coil.decode
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Paint
+import android.media.MediaMetadataRetriever
+import android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT
+import android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION
+import android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH
+import android.media.MediaMetadataRetriever.OPTION_CLOSEST_SYNC
+import android.os.Build.VERSION.SDK_INT
+import androidx.core.graphics.applyCanvas
+import androidx.core.graphics.drawable.toDrawable
+import coil.bitmap.BitmapPool
+import coil.request.videoFrameMicros
+import coil.request.videoFrameOption
+import coil.size.OriginalSize
+import coil.size.PixelSize
+import coil.size.Size
+import kotlin.math.roundToInt
+
+internal class VideoFrameDecoderDelegate(private val context: Context) {
```

Comment:
This class is basically pulled out entirely from `VideoFrameFetcher` without changes.